src/jit/scopeinfo.cpp

```diff
@@ -154,6 +154,92 @@ CodeGenInterface::siVarLoc::siVarLoc(const LclVarDsc* varDsc, regNumber baseReg,
     }
 }
 
+//------------------------------------------------------------------------
+// Equals: Compares first reference and then values of the structures.
+//
+// Arguments:
+//    lhs   - a "siVarLoc *" to compare.
+//    rhs   - a "siVarLoc *" to compare.
+//
+// Notes:
+//    Return true if both are nullptr.
+//
+// static
+bool CodeGenInterface::siVarLoc::Equals(const siVarLoc* lhs, const siVarLoc* rhs)
+{
+    if (lhs == rhs)
+    {
+        // Are both nullptr or the same reference
+        return true;
+    }
+    if ((nullptr == lhs) || (nullptr == rhs))
+    {
+        // Just one of them is a nullptr
+        return false;
+    }
+
+    bool areEquals = true;
```

Comment:
Maybe I am missing something but it looks like if `(lhs->vlType != rhs->vlType)`  then we return true:
`bool areEquals = true;`?

Reply:
Yes, sorry for that